xarray/core/dataarray.py

```diff
@@ -1377,13 +1378,15 @@ def func(self, *args, **kwargs):
         return func
 
     @staticmethod
-    def _binary_op(f, reflexive=False, join='inner', **ignored_kwargs):
+    def _binary_op(f, reflexive=False, join=None, **ignored_kwargs):
         @functools.wraps(f)
         def func(self, other):
             if isinstance(other, (Dataset, groupby.GroupBy)):
                 return NotImplemented
             if hasattr(other, 'indexes'):
-                self, other = align(self, other, join=join, copy=False)
+                # if user does not specify join, default to OPTIONS['join']
+                how_to_join = join or OPTIONS['join']
```

Comment:
You could do the `if`/`else` inline, if you prefer:
`how_to_join = OPTIONS['join'] if join is None else join`

But yes, I do prefer explicitly checking against `None` for default values rather than relying on implicit falseness of `None`.